spec/rspec/support/object_inspector_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+require 'rspec/support/object_inspector'
+require 'rspec/matchers/fail_matchers'
+
+module RSpec
+  module Support
+    describe ObjectInspector, ".inspect" do
+      include RSpec::Matchers::FailMatchers
+
+      context 'with Time objects' do
+        let(:time) { Time.utc(1969, 12, 31, 19, 01, 40, 101) }
+        let(:formatted_time) { ObjectInspector.inspect(time) }
+
+        it 'produces an extended output' do
+          expected_output = "1969-12-31 19:01:40.000101"
+          expect(formatted_time).to include(expected_output)
+        end
+      end
+
+      context 'with DateTime objects' do
+        def with_date_loaded
+          in_sub_process_if_possible do
+            require 'date'
+            yield
+          end
+        end
+
+        let(:date) { DateTime.new(2000, 1, 1, 1, 1, Rational(1, 10)) }
+        let(:formatted_date) { ObjectInspector.inspect(date) }
+
+        it 'produces a descript output using insepct' do
+          with_date_loaded do
+            expect(formatted_date).to eq(date.inspect)
+          end
+        end
+
+        it 'does not require DateTime to be defined since you need to require `date` to make it available' do
+          hide_const('DateTime')
+          expect(ObjectInspector.inspect('Test String')).to eq('"Test String"')
+        end
+
+        context 'when ActiveSupport is loaded' do
+          it "uses a custom format to ensure the output is different when DateTimes differ" do
+            stub_const("ActiveSupport", Module.new)
+
+            with_date_loaded do
+              expected_date = date.strftime("%a, %d %b %Y %H:%M:%S.%N %z")
+              expect(formatted_date).to eq(expected_date)
```

Comment:
This expectation simply mirrors the implementation.  It'd be good to make `expected_date` be a known, static value.  Is that feasible?